Source/items.cpp

```diff
@@ -3219,9 +3219,19 @@ Item *SpawnUnique(_unique_items uid, Point position, bool sendmsg /*= true*/)
 	while (AllItemsList[idx].iItemId != UniqueItems[uid].UIItemId)
 		idx++;
 
-	GetItemAttrs(item, static_cast<_item_indexes>(idx), curlv);
-	GetUniqueItem(*MyPlayer, item, uid);
-	SetupItem(item);
+	if (!gbIsMultiplayer || FlipCoin(15)) {
+		GetItemAttrs(item, static_cast<_item_indexes>(idx), curlv);
+		GetUniqueItem(*MyPlayer, item, uid);
+		SetupItem(item);
+	} else {
+		if (level)
+			curlv = *level;
+		const ItemData &uniqueItemData = AllItemsList[idx];
+		_item_indexes idx = GetItemIndexForDroppableItem(false, [&uniqueItemData](const ItemData &item) {
+			return item.itype == uniqueItemData.itype;
+		});
+		SetupAllItems(*MyPlayer, item, idx, AdvanceRndSeed(), curlv * 2, 15, true, false, false);

```

Comment:
Would it work to set a 0% chance of unique so that the unique is the quest item?
```suggestion
		SetupAllItems(*MyPlayer, item, idx, AdvanceRndSeed(), curlv * 2, 0, true, false, false);
```

Reply:
When changing `uper` to 0, we will also reduce the item base level `GetItemBLevel` and affect magical items.
Also the result could still be a unique, cause `GenerateRnd` could return 0 and the check in `CheckUnique` would still succeed (but quite unlikely).

Another idea could be to put `SetupAllItems` (inklusive `AdvanceRndSeed`) in a loop and run the loop until the item is not a unique.

Personally I wouldn't mind if another unique item is dropped/rewarded.

Reply:
Ok, would 1 also have negative side effects on the generated item? I'm ok with it potentially being another item, but preferable a very low chance.